sys/riscv/riscv/pmap.c

```diff
@@ -4126,8 +4126,36 @@ pmap_caploadgen_update(pmap_t pmap, vm_offset_t va, vm_page_t *mp, int flags)
 
 	return res;
 }
-#endif
-#endif
+
+void
+pmap_assert_consistent_clg(pmap_t pmap, vm_offset_t va)
+{
+	pt_entry_t *l2, l2e;
+	pt_entry_t *pte, tpte;
+
+	l2 = pmap_l2(pmap, va);
+	if (l2 == NULL || ((l2e = pmap_load(l2)) & PTE_V) == 0)
+		return;
+	if ((l2e & PTE_RWX) != 0) {
+		/* Large page */
+		pte = l2;
```

Comment:
Could drop this and rename pte to l3 (and maybe combine with the l2 line instead for the definition), the code only uses tpte for this path